### PR TITLE
Association

### PR DESCRIPTION
**Pull request by Olamarx**
- Create a class Classroom with the following:
   @label instance variable, should be initialized in the constructor.
   Setter and getter for @label (remember about attr_accessor).
-Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
  Make sure that when adding a student to a classroom it also sets the classroom for the student.
  Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
  @title and @author instance variables, should be initialized in the constructor.
  Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
  @date instance variable, should be initialized in the constructor.
  Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
  Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental=(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (Mihndim2020) [inline book.rb:14]: - [OPTIONAL] The `person and date` should be passed in as parameters to the `add_rental` method, something like so: 
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```
Please kindly fix that.

File: person.rb [hunk 2]
```diff
@@ -23,16 +24,14 @@ def can_use_service?
     of_age? || @parent_permission
   end
 
+  def add_tental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (Mihndim2020) [inline person.rb:30]: - [OPTIONAL] The `book and date` should be passed in as parameters to the `add_rental` method, something like so: 
```
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```
Please kindly fix that.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
### Hi @Olamarx, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Olamarx) [reply book.rb:14]: @Mihndim2020 thank you very much, I have passed them as parameters.

File: person.rb (hunk 2 above)
Comment (Olamarx) [reply person.rb:30]: @Mihndim2020 Thank you very much, I have passed them as parameters.

**Pull request merged**